### ISSUE TITLE
Improve error message for missing multiplication sign

### ISSUE THREAD

**Issue by eschnett** [JuliaLang/julia#40064]
When I forget a multiplication sign `*` between two parentheses, I receive this error:
```
julia> (1/2)(3+4)
ERROR: MethodError: objects of type Float64 are not callable
```

This error is confusing for beginners. (A colleague was just stuck with that error and asked for help.) Could you add "Maybe you forgot a multiplication operator `*` after a number?" to it?

**Comment by simeonschaub** [comment]
Seems like a pretty good use case for `Base.Experimental.register_error_hint` in case anyone wants to tackle this.

**Comment by ArchieCall** [comment]
Since Julia does not know the actual desired operator, the hint could be something like: "Maybe you forgot an operator such as:  ( * | ^ | % | / | ...)".

**Comment by simeonschaub** [comment]
I think the confusion here is more that we do actually allow `2(3+4)` or `a=3+4; (1/2)a` for multiplication, but not the example mentioned here.

**Comment by eschnett** [comment]
Other languages (Mathematica) as well as common mathematical notation allow the expression above, and infer the multiplication. I think that suggesting a missing multiplication is most helpful to beginners.

**Comment by Sov-trotter** [comment]
I seem to be stuck while trying this out. 
`Base.Experimental.register_error_hint(handler, type)` needs the handler to be defined in the module.
In our case where the module is Base, there seems no direct way to pass `Base.mul_int` into the handler. In other words there isn't any concrete way to find out that the error has been thrown by `Base.mul_int`. 

Eg: the example [here](https://docs.julialang.org/en/v1/base/base/#Base.Experimental.register_error_hint) uses  
`if exc.f == only_int` if exception has been generated by the `only_int` method.
 
Any help would be appreciated!

**Comment by simeonschaub** [comment]
The error is not thrown by `Base.mul_int`, but instead by trying to call an instance of a number, so you would need to check for `if exc.f isa Number` im the handler.

**Comment by Sov-trotter** [comment]
Oh. Okay. 
But that would throw the hint every time `exc.f isa Number` which is not specific for the issue that we are trying to tackle?

**Comment by simeonschaub** [comment]
Yes, it would be shown every time you try to call a number (i.e. the function is a number, it wouldn't be be shown if only an argument were a number), but that is what we want here.

**Comment by Sov-trotter** [comment]
Got it. Thanks!
